ratify-project/ratify
# test: Add e2e test for CRL

# Description

## What this PR does / why we need it:
This PR added CRL e2e test
- CRL CLI e2e test with 
  - happy path: verify with cert support CRL
  - negative path: verify with revoked CRL
- CRL K8S e2e test
- Manual testing signing and verify output:
  1.  Image Signed by cert with CRL
![image](https://github.com/user-attachments/assets/87c43c56-2773-4562-982e-50cc548c05db)
  2.  Image Verified 
![image](https://github.com/user-attachments/assets/4cada024-6ad0-460d-aeb6-7759ab0dc4fb)


Notation e2e test has covered following scenarios and we are not going to go over them.
1. successfully completed revocation check with CRL cache
2. failed revocation check with revoked leaf certificate
3. failed revocation check with revoked intermediate certificate
4. successfully completed with cache creation error in warning message
5. successfully completed with the crl in the cache expired and a roundtrip
6. failed with crl in the cache expired and a roundtrip to download a revoked crl

## Which issue(s) this PR fixes *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
N/A

## Type of change

Please delete options that are not relevant.

- [ ] Helm Chart Change (any edit/addition/update that is necessary for changes merged to the `main` branch)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] CI test


# Checklist:

- [ ] Does the affected code have corresponding tests?
- [ ] Are the changes documented, not just with inline documentation, but also with conceptual documentation such as an overview of a new feature, or task-based documentation like a tutorial? Consider if this change should be announced on your project blog.
- [ ] Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ] Do all new files have appropriate license header?

# Post Merge Requirements
- [ ] MAINTAINERS: manually trigger the "Publish Package" workflow after merging any PR that indicates `Helm Chart Change`
